website/docs/e/coi_with_compute_instance.html.markdown

```diff
@@ -0,0 +1,229 @@
+ 
+First create a Terraform config file named ```main.tf```. Inside, you'll want to include the configuration of 
+[Yandex.Cloud Provider](https://www.terraform.io/docs/providers/yandex/index.html), 
+[compute instance](https://www.terraform.io/docs/providers/yandex/r/compute_instance.html)
+and [compute image](https://www.terraform.io/docs/providers/yandex/d/datasource_compute_image.html).
+
+Use Yandex provider:
+
+```hcl
+provider "yandex" {
+  token     = "your YC_TOKEN"
+  folder_id = "your folder id"
+  zone      = "your default zone"
+}
+```
+
+Configure Yandex provider:
+
+* The `token` field should be replaced with your personal Yandex.Cloud authentication token.
+* The `folder` field is the id of your folder to create Container Optimized Image.
+* The `zone` field should be replaced with default [availability zone]((https://cloud.yandex.com/docs/overview/concepts/geo-scope)) to operate under.
+
+Use already created Container Optimized Image from [image family](https://cloud.yandex.com/docs/compute/concepts/images#family) collection :
+
+```hcl
+data "yandex_compute_image" "container-optimized-image" {
+  family    = "container-optimized-image"
+}
+```
+
+Create compute instance:
+
+```hcl
+resource "yandex_compute_instance" "instance-based-on-coi" {
+
+  boot_disk {
+    initialize_params {
+      image_id = data.yandex_compute_image.container-optimized-image.id
+    }
+  }
+  network_interface {
+    subnet_id = "your subnet id"
+    nat       = true
+  }
+  resources {
+    cores  = 2
+    memory = 2
+  }
+
+  metadata = {
+    docker-container-declaration = file("${path.module}/declaration.yaml")
+    user-data                    = file("${path.module}/cloud_config.yaml")
+  }
+}
+```
+
+Configure compute instance:
+
+* The `subnet_id` field is the id of your virtual private cloud [subnet](https://www.terraform.io/docs/providers/yandex/d/datasource_vpc_subnet.html).
+
+Create a cloud specification file named  ```cloud-config.yaml``` and put it to the same folder:
+
+```hcl
+#cloud-config.yaml
+ssh_pwauth: no
+users:
+  - name: yc-user
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    shell: /bin/bash
+    ssh_authorized_keys:
+      - "your public ssh key"
+```
+
+Configure cloud specification:
+
+* Fill the `ssh_authorized_keys` value with your public ssh key.
+
+Create Container Optimized Image specification file named ```declaration.yaml``` and put it to the same folder:
+
+```hcl
+#declaration.yaml
+spec:
+  containers:
+  - image: cr.yandex/yc/demo/coi:v1
+    securityContext:
+      privileged: false
+    stdin: false
+    tty: false
+```
+
+Create ```output.tf``` file to get IP address of the Container Optimized Image:
+
+```hcl
+output "external_ip" {
+  value = yandex_compute_instance.instance-based-on-coi.network_interface.0.nat_ip_address
+}
+```
+
+## Launching Container Optimized Image
+
+Now everything is set to launch the COI instance in Terraform. Make following list of instructions:
```

Comment:
Should rephrase: 'Make following list of instructions:'
'Execute the following list of instructions:' ?
'Proceed the following list of instructions:' ?
'Follow the steps described below:' ?